src/client/datascience/notebook/notebookSerializer.ts

```diff
@@ -27,7 +27,9 @@ export class NotebookSerializer implements VSCNotebookSerializer {
         @inject(NotebookCellLanguageService) private readonly cellLanguageService: NotebookCellLanguageService
     ) {}
     public deserializeNotebook(content: Uint8Array, _token: CancellationToken): NotebookData {
-        const contents = Buffer.from(content).toString();
+        let contents = new TextDecoder().decode(content.buffer);
+		// Decoded string appears to contain $dataToNotebook + unicode junk? Skip past the junk
```

Comment:
I wonder why Buffer skips it? Shouldn't it have the same problem?

Reply:
Is there a start index perhaps on the content?

Reply:
It might be a web worker extension host problem moreso than a TextDecoder vs Buffer problem. $dataToNotebook is a function in VS Code source

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/byteOffset?